crates/eips/src/eip7702/auth_list.rs

```diff
@@ -0,0 +1,119 @@
+use core::ops::Deref;
+
+#[cfg(not(feature = "std"))]
+use alloc::vec::Vec;
+use alloy_primitives::{keccak256, Address, ChainId, Signature, SignatureError, B256};
+use alloy_rlp::{BufMut, Decodable, Encodable, Header, RlpEncodable};
+
+use super::constants::MAGIC;
+
+/// An EIP-7702 authorization list.
+pub type AuthorizationList = Vec<Authorization>;
+
+/// An EIP-7702 authorization.
+#[derive(Debug, Clone, RlpEncodable)]
+pub struct Authorization {
+    chain_id: ChainId,
+    address: Address,
+    nonce: OptionalNonce,
```

Comment:
these fields are the raw authorization, right?

so would it make sense to have this as a standalone type? because I guess this authorization is then signed?

similar to Tx and the Signed type

Reply:
makes sense

Reply:
a bit annoying, can't reuse `Signed<T>` because it's in `alloy-consensus`, will take a gander though

Reply:
I split it into a signed/unsigned type, and I also made it generic over the signature in line with the transaction type. The only hurdle here is that we cannot default `Sig` to `Signature`. wdyt?